packages/web3-core-types/src/Hex/Hex.js

```diff
@@ -0,0 +1,378 @@
+/*
+    This file is part of web3.js.
+
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file Hex.js
+ * @author Oscar Fonseca <hiro@cehh.io>
+ * @date 2019
+ */
+
+import {isNumber, isString} from 'lodash';
+import utf8 from 'utf8';
+
+export default class Hex {
+    /**
+     * @dev Wrap as object
+     * @param {String} hex
+     *
+     * @constructor
+     */
+    constructor(params) {
+        const requires = ['hex'];
+
+        this.props = {};
+
+        if (isString(params)) {
+            params = {
+                hex: params
+            };
+        }
+
+        if (Hex.isValid(params.hex)) {
+            this.props.hex = params.hex.toString();
+        }
+
+        if (params.hex === 'empty') {
+            this.props.hex = '0x';
+        }
+
+        requires.forEach((propName) => {
+            if (typeof this.props[propName] === 'undefined') {
+                this._throw(propName, params[propName]);
+            }
+        });
+
+        /* Make the props immutable */
+        Object.freeze(this.props);
+    }
+
+    /**
+     * Check if the supplied string is a valid hex value
+     *
+     * @method isValid
+     *
+     * @param {string} parameter
+     *
+     * @return {boolean}
+     */
+    static isValid(hex) {
+        return /^(-)?(0x)?[0-9a-fA-F]*$/.test(hex);
+    }
+
+    /**
+     * Check if the supplied string is hex with 0x prefix
+     *
+     * @method isStrict
+     *
+     * @param {string} hex
+     *
+     * @return {boolean}
+     */
+    static isStrict(hex) {
+        return /^(-0x|0x)[0-9a-fA-F]*$/.test(hex);
+    }
+
+    /**
+     * Build an object of Hex from a string
+     *
+     * @method fromString
+     *
+     * @param {String} value
+     *
+     * @returns {Hex}
+     */
+    static fromString(value) {
+        if (!isString(value)) throw new Error(`The given value ${value} is not string type.`);
+
+        value = value.replace(/(-)?(0x)?([0-9a-f]*)/i, '$10x$3');
+
+        const params = {
+            hex: value
+        };
+
+        return new Hex(params);
+    }
+
+    /**
+     * Build an object of Hex from a base 10 number
+     *
+     * @method fromNumber
+     *
+     * @param {Number} value
+     *
+     * @returns {Hex}
+     */
+    static fromNumber(value) {
+        if (!isNumber(value)) throw new Error(`The given value ${value} is not number type.`);
```

Comment:
I'm not a big fan of shorthands because the bundler will improve it anyways later. :-)